pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -94,34 +104,92 @@ func (s *nodeTopoInformer) Setup(ctx *pluginOption, state *pluginState) {
 		klog.Fatalf("pods informer format error")
 	}
 	s.podsInformer = podsInformer
+
+	s.NodeResourceTopologyInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			nodeResourceTopology, ok := obj.(*topov1alpha1.NodeResourceTopology)
+			if ok {
+				s.updateNodeTopo(nodeResourceTopology)
+			} else {
+				klog.Errorf("node topology informer add func parse failed")
+			}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			oldNodeResourceTopology, oldOK := oldObj.(*topov1alpha1.NodeResourceTopology)
+			newNodeResourceTopology, newOK := newObj.(*topov1alpha1.NodeResourceTopology)
+			if !oldOK || !newOK {
+				klog.Errorf("unable to convert object to NodeResourceTopology, old %T, new %T", oldObj, newObj)
+				return
+			}
+			if !isSyncNeeded(oldNodeResourceTopology, newNodeResourceTopology, ctx.NodeName) {
+				klog.V(5).Infof("find NodeResourceTopology %s has not changed.", newNodeResourceTopology.Name)
+				return
+			}
+			klog.Infof("update node Resource Topology %s", newNodeResourceTopology.Name)
+			s.updateNodeTopo(newNodeResourceTopology)
+		},
+	})
+
 }
 
 func (s *nodeTopoInformer) Start(stopCh <-chan struct{}) {
 	klog.V(2).Infof("starting node topo informer")
-	if !cache.WaitForCacheSync(stopCh, s.nodeInformer.HasSynced, s.podsInformer.HasSynced) {
+	if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
+		go s.NodeResourceTopologyInformer.Run(stopCh)
+	}
+	if !cache.WaitForCacheSync(stopCh, s.NodeResourceTopologyInformer.HasSynced, s.nodeInformer.HasSynced, s.podsInformer.HasSynced) {
```

Comment:
what if the feature-gate `NodeTopologyReport=false`, would the `s.NodeResourceTopologyInformer.HasSynced()` become true?